proposals/nnnn-screenmanager-push-pop.md

```diff
@@ -0,0 +1,129 @@
+ */
+- (void)popTemplate;
+
+/**
+ The current template will deactivate and the template stack will be cleared except for the first item, which will either activate or be replaced if a new template is passed. That new template will then activate. If there is only one object in the templateStack and no newRootTemplate is passed, nothing will happen.
+ */
+- (void)popToRootTemplate:(nullable id<SDLTemplate>)newRootTemplate;
+
+/**
+ The current template will deactivate. Then the new template will activate by replacing the current template on the stack instead of pushing it onto the stack.
+ */
+- (void)replaceCurrentTemplateWithNewTemplate:(id<SDLTemplate>)newTemplate;
+
+/**
+ All text, images, and soft buttons will be cleared from the screen. Preloaded choices and main menu items will remain.
+ */
+- (void)clearScreen;
+
+@end
+```
+
+### Additional Implementation Notes
+1. When connected to systems running RPC <6.0, the screen manager should send `SetDisplayLayout` to change the template, and on systems running RPC >6.0, the screen manager should send `Show` to change the template.
+
+### Java APIs
+This will be set up in a similar way to the Obj-C APIs above. The `SDLTemplate` protocol and screen manager additions above would be translated into appropriate Java APIs. Any necessary changes are at the discretion of the project maintainers. However larger changes that would impact back into the Objective-C code above (such as adding or removing a method) will require proposal revisions.
+
+### JavaScript APIs
+The JavaScript APIs will be set up in a similar way to the Obj-C APIs above. The `SDLTemplate` protocol and screen manager additions above would be translated into appropriate JavaScript APIs. All changes will be at the discretion of the project maintainers. However larger changes that would impact back into the Objective-C code above (such as adding or removing a method) will require proposal revisions.
+
+## Potential downsides
+
+This proposal adds additional complexity to the manager layer to handle changing templates. This is a moderate source of complexity, however, this complexity burden is currently placed on each developer. Furthermore, this should enable developers to build more maintainable apps by separating out their code related to their individual template screens.
+
+## Impact on existing code
+
+This would be a minor version update to all libraries implementing a screen manager, namely, the iOS, Java, and JavaScript app libraries.
+
+## Alternatives considered
+
+1. We could only have an API to change the template and remove the push/pop and separation of screens APIs, but the author determined that the change to allow developers to separate their code was a big improvement for developers who can now easily separate their unrelated SDL code into contained classes.
```

Comment:
Should "screens APIs" be changed to "screen's APIs" or "screens' APIs" or "screen APIs"?